### PR TITLE
Re-adds room beauty without the negative moodlets

### PR DESCRIPTION
**Pull request by ghost**
:cl: Floyd / Qustinnus
 add: You can now appreciate rooms for being very well decorated.
/:cl:

This system was in once when mood was very new, but the bad moodlets were very overpowering and didn't make it fun, hopefully without the bad moodlets thisll work better.

for clarification, this is area based. Due to limitations in byond we can't sanely check your surroundings every few seconds so its more a general update on what the room looks like.

### CODE REVIEW

**Comment by Krysonism** [comment]
message_admins("small penis")

**Comment by Qustinnus** [comment]
oh yeah i forgot my debug stuff

**Comment by Mickyan** [comment]
shouldn't this be tied directly to the art component? For instance tagger graffiti counts as good art but doesn't currently affect beauty

**Comment by Qustinnus** [comment]
Good point I'll check but I'm keeping em seperate

File: code/datums/components/mood.dm [hunk 1]
```diff
@@ -353,11 +353,34 @@
...
+	if(A.outdoors) //if we're outside, we don't care.
+		clear_event(null, "area_beauty")
+		return FALSE
+	if(HAS_TRAIT(parent, TRAIT_SNOB))
+		switch(A.beauty)
+			if(-INFINITY to BEAUTY_LEVEL_HORRID)
+				add_event(null, "area_beauty", /datum/mood_event/horridroom)
+				return
+			if(BEAUTY_LEVEL_HORRID to BEAUTY_LEVEL_BAD)
+				add_event(null, "area_beauty", /datum/mood_event/badroom)
```
Comment (nemvar) [inline code/datums/components/mood.dm]: Maybe always apply the moodlet but without the negative mood penalty just so people can get some feedback on what is considered good/bad.

File: code/datums/components/beauty.dm [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	var/beauty = 0
+
+/datum/component/beauty/Initialize(beautyamount)
+	if(!ismovableatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	beauty = beautyamount
+	RegisterSignal(parent, COMSIG_ENTER_AREA, .proc/enter_area)
+	RegisterSignal(parent, COMSIG_EXIT_AREA, .proc/exit_area)
+	var/area/A = get_area(parent)
+	if(!A || A.outdoors)
```
Comment (nemvar) [inline code/datums/components/beauty.dm]: This check does not exist on the enter_area proc

File: code/datums/components/beauty.dm [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+		return
+	A.totalbeauty += beauty
```
Comment (nemvar) [inline code/datums/components/beauty.dm]: Why the duplicate code? Just call enter_area.

File: code/datums/components/beauty.dm [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	A.totalbeauty -= beauty
+	A.update_beauty()
+
+/datum/component/beauty/Destroy()
+	. = ..()
+	var/area/A = get_area(parent)
+	if(!A || A.outdoors)
+		return
+	A.totalbeauty -= beauty
+	A.update_beauty()
```
Comment (nemvar) [inline code/datums/components/beauty.dm]: Again, just call exit_area

File: code/datums/materials/_material.dm [hunk 5]
```diff
@@ -38,6 +40,9 @@ Simple datum which is instanced once per type and is used for every object of sa
 	if(material_flags & MATERIAL_ADD_PREFIX)
 		source.name = "[name] [source.name]"
 
+	if(beauty_modifier)
+		addtimer(CALLBACK(source, /datum.proc/AddComponent, /datum/component/beauty, beauty_modifier * amount), 0)
```
Comment (nemvar) [inline code/datums/materials/_material.dm]: Why exactly does this need to be async?

Comment (Qustinnus) [reply code/datums/materials/_material.dm]: honest to god I just got this from the OG implementation from back in the day and I remember me getting stuck on this for ages and being told by ninja or something that i should make it async.

File: code/datums/components/beauty.dm (hunk 2 above)
Comment (nemvar) [reply code/datums/components/beauty.dm]: The !A makes sense, but the outdoors thing should probably still be in there.

File: code/datums/mood_events/beauty_events.dm [hunk 6]
```diff
@@ -0,0 +1,19 @@
+/datum/mood_event/horridroom
+	description = "<span class='boldwarning'>This room looks terrible!</span>\n"
+	mood_change = -5
+
+/datum/mood_event/badroom
+	description = "<span class='warning'>This room looks really bad.</span>\n"
+	mood_change = -3
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/mood_events/beauty_events.dm]: 😏

Comment (Qustinnus) [reply code/datums/mood_events/beauty_events.dm]: quirk 😏

**Pull request merged**